### ISSUE TITLE
provide .well-known endpoint for oidc

### ISSUE THREAD

**Issue by ymajoros** [badgateway/oauth2-client#67]
For openid-connect, the spec specifies that the endpoint should be `/.well-known/openid-configuration`

This conflicts with the currently default path, which seems to suit OAuth2 well though.

I suggest adding a flag to choose between OAuth2 only or OIDC.

**Comment by evert** [comment]
Does setting the `discoveryEndpoint` currently work for you? I assume this request is only to make it easier to 'guess' the openid endpoint, right?

**Comment by ymajoros** [comment]
Yes, it works (but see other reported bugs). This is just a feature request for convenience.
